### PR TITLE
lazyLoad MultiRecord View

### PR DESCRIPTION
**Pull request by kebeclibre**
Display viewtype from url

loadState with lazyController: (a multiRecord view if we asked for a single record view)

### CODE REVIEW

File: addons/wowl/static/src/services/action_manager/action_manager.ts [hunk 1]
```diff
@@ -470,18 +477,36 @@ function makeActionManager(env: OdooEnv): ActionManager {
       const searchViewId = action.search_view_id ? action.search_view_id[0] : false;
       action.views.push([searchViewId, "search"]);
     }
-    // TODO: take the view type present in options.
-    // TODO: if view !== multiview, push a multiview before it
-    const view = views[0];
+    let view = options.viewType && views.find((v) => v.type === options.viewType);
+    let lazyController: ViewController | undefined;
+    if (view && !view.multiRecord) {
+      const lazy = views[0] && views[0].multiRecord ? views[0] : undefined;
```
Comment (aab-odoo) [inline addons/wowl/static/src/services/action_manager/action_manager.ts]: views[0] exists here, doesn't it?

Comment (aab-odoo) [reply addons/wowl/static/src/services/action_manager/action_manager.ts]: rename to lazyView, maybe ?

File: addons/wowl/static/src/services/action_manager/action_manager.ts [hunk 2]
```diff
@@ -786,11 +811,19 @@ function makeActionManager(env: OdooEnv): ActionManager {
...
+      index = controllerStack.findIndex(
+        (ct) => ct.action.jsId === controller.action.jsId && (ct as ViewController).view.multiRecord
+      );
       index = index > -1 ? index : controllerStack.length - 1;
+    } else {
+      index = controllerStack.findIndex(
+        (ct) =>
+          ct.action.jsId === controller.action.jsId && !(ct as ViewController).view.multiRecord
+      );
+      index = index > -1 ? index : controllerStack.length;
```
Comment (aab-odoo) [inline addons/wowl/static/src/services/action_manager/action_manager.ts:826]: not sure about the logic here

File: addons/wowl/static/src/services/action_manager/action_manager.ts [hunk 3]
```diff
@@ -864,17 +896,17 @@ function makeActionManager(env: OdooEnv): ActionManager {
...
       }
     } else if (state.model && (state.view_type || state.id)) {
       if (state.id) {
         action = {
           res_model: state.model,
-          res_id: state.id,
+          res_id: parseInt(state.id, 10),
           type: "ir.actions.act_window",
-          views: [[state.view_id || false, "form"]],
+          views: [[state.view_id ? parseInt(state.view_id) : false, "form"]],
```
Comment (aab-odoo) [inline addons/wowl/static/src/services/action_manager/action_manager.ts]: missing base

File: addons/wowl/static/src/services/action_manager/action_manager.ts [hunk 4]
```diff
@@ -349,7 +351,8 @@ function makeActionManager(env: OdooEnv): ActionManager {
    */
   async function _updateUI(
     controller: Controller,
-    options: UpdateStackOptions = {}
+    options: UpdateStackOptions = {},
+    lazyController?: Controller
```
Comment (aab-odoo) [inline addons/wowl/static/src/services/action_manager/action_manager.ts]: maybe pass this as an option?

**Review by aab-odoo — COMMENTED** [review]
cool :-)

File: addons/wowl/static/src/services/action_manager/action_manager.ts [hunk 5]
```diff
@@ -34,6 +34,7 @@ export type ActionRequest = ActionId | ActionXMLId | ActionTag | ActionDescripti
 export interface ActionOptions {
   clearBreadcrumbs?: boolean;
   viewType?: ViewType;
+  resID?: number;
```
Comment (aab-odoo) [inline addons/wowl/static/src/services/action_manager/action_manager.ts]: convention in our branch: resId, I think

**Review by aab-odoo — APPROVED** [review]

File: addons/wowl/static/src/services/action_manager/action_manager.ts (hunk 5 above)
Comment (kebeclibre) [reply addons/wowl/static/src/services/action_manager/action_manager.ts]: Done, @aab-odoo

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**